### PR TITLE
[ESLINT] Client ES Linting error Fixes for rule 'jsx-no-bind' in the Comments related files

### LINKED ISSUE #5288: [ESLINT] Client ES Linting error Fixes for rule 'jsx-no-bind' in the Comments related files
ESLint Rule "react/jsx-no-bind" needs to be added to the client-side codebase to potentially decrease unnecessary Re-renders and improve performance.

As part of that Effort, this issue is created to track the progress of the rule related fixes in the following folders,
- appsmith/app/client/src/comments/

### PR DESCRIPTION
**Pull request by rahulramesha**
## Description

ESLint Rule jsx-no-bind Rule Fixes for Comment section

Fixes #5288 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)(Code ESlint issue)

## How Has This Been Tested?

>Manual verification

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: ISSUE/5210_ESlint_Rules_Update 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 53.39 **(0.01)** | 35.48 **(0.08)** | 32 **(-0.01)** | 53.92 **(0.01)**
 :red_circle: | app/client/src/comments/AppComments/AppCommentsFilterPopover.tsx | 42.5 **(-2.24)** | 66.67 **(0)** | 0 **(0)** | 44.74 **(-2.48)**
 :green_circle: | app/client/src/comments/CommentCard/CommentCard.tsx | 75.61 **(0.3)** | 69.92 **(0.76)** | 43.9 **(0)** | 76.32 **(0.32)**
 :green_circle: | app/client/src/comments/CommentCard/CommentContextMenu.tsx | 90.24 **(10.65)** | 75 **(-6.82)** | 66.67 **(33.34)** | 92.11 **(11.68)**
 :sparkles: :new: | **app/client/src/comments/CommentCard/CommentContextMenuOption.tsx** | **41.18** | **100** | **0** | **46.67**
 :green_circle: | app/client/src/comments/CommentThread/CommentThread.tsx | 92 **(0.16)** | 61.29 **(2.03)** | 80 **(0)** | 93.62 **(-1.94)**
 :green_circle: | app/client/src/comments/inlineComments/InlineCommentPin.tsx | 85.25 **(0.25)** | 61.36 **(1.36)** | 85.71 **(0)** | 85.96 **(0.25)**
 :green_circle: | app/client/src/comments/inlineComments/UnpublishedCommentThread.tsx | 89.29 **(0.4)** | 50 **(0)** | 71.43 **(0)** | 88.46 **(0.46)**</details>

### CODE REVIEW

**Comment by rishabhsaxena** [comment]
@rahulramesha 
Actually `useCallback` can add overheads in cases where we don't need referential equality. But I guess using them as a default does help in, us not having to deal with perf cases manually. 

Ref: https://kentcdodds.com/blog/usememo-and-usecallback#why-is-usecallback-worse

**Review by rishabhsaxena — APPROVED** [review]

**Comment by rahulramesha** [comment]
@rishabhsaxena I understand where the author is coming from. But I feel that the javascript side of the operation with useCallback is relatively inexpensive compared to the potential re-renders which might be caused.

File: app/client/src/comments/CommentCard/CommentContextMenuOption.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  onClick: typeof noop;
+}
+
+type Props = {
+  handleClick: typeof noop;
+  option: Option;
+};
+
+function CommentContextMenuOption({ handleClick, option }: Props) {
+  const onClickHandler = useCallback(() => handleClick(option), [handleClick]);
```
Comment (SatishGandham) [inline app/client/src/comments/CommentCard/CommentContextMenuOption.tsx:41]: Shouldn't this depend on both option and handleclick?

Comment (rahulramesha) [reply app/client/src/comments/CommentCard/CommentContextMenuOption.tsx:41]: Yes it should, don't know how I missed that

**Comment by rahulramesha** [comment]
Closing this as it was decided to not take this up now

**Pull request closed without merging**